test/integration/link.cc

```diff
@@ -351,6 +351,35 @@ TEST_F(LinkIntegrationTest, LinkAccelerations)
       ecm.Component<components::WorldAngularAcceleration>(eLink));
 }
 
+//////////////////////////////////////////////////
+TEST_F(LinkIntegrationTest, LinkInertia)
+{
+  EntityComponentManager ecm;
+  EventManager eventMgr;
+  SdfEntityCreator creator(ecm, eventMgr);
+
+  auto eLink = ecm.CreateEntity();
+  ecm.CreateComponent(eLink, components::Link());
+
+  Link link(eLink);
+  EXPECT_EQ(eLink, link.Entity());
+
+  ASSERT_TRUE(link.Valid(ecm));
+
+  // Before we add components, pose functions should return nullopt
+  EXPECT_EQ(std::nullopt, link.WorldInertial(ecm));
+
+  math::MassMatrix3d linkMassMatrix(10.0, {0.4, 0.4, 0.4}, {0.02, 0.02, 0.02});
+  math::Pose3d linkWorldPose;
+  linkWorldPose.Set(0.0, 0.1, 0.2, 0.0, GZ_PI_4, GZ_PI_2);
```

Comment:
When setting the `Inertial` component on a link, the pose of the inertial is relative to the link frame. Using the variable name `linkWorldPose` is slightly confusing since it appears that we're trying to set a world pose on the `Inertial` object. The test only works since the `pose` of the link entity itself is identity. In fact, it would be great to set the pose of `link` to something other than identity to check that the proper world inertial is being calculated.

Reply:
Done. PTAL https://github.com/gazebosim/gz-sim/pull/2218/commits/5f107009bfa2d8fd468f5a994271386404c77032